src/pybind/mgr/nfs/export.py

```diff
@@ -836,45 +845,6 @@ def _apply_export(
             log.debug('export %s pseudo %s -> %s',
                       new_export.export_id, old_export.pseudo, new_export.pseudo)
 
-        if old_export.fsal.name == NFS_GANESHA_SUPPORTED_FSALS[0]:
-            old_fsal = cast(CephFSFSAL, old_export.fsal)
-            new_fsal = cast(CephFSFSAL, new_export.fsal)
-            if old_fsal.user_id != new_fsal.user_id:
-                self._delete_export_user(old_export)
-                self._create_export_user(new_export)
-            elif (
-                old_export.path != new_export.path
-                or old_fsal.fs_name != new_fsal.fs_name
-            ):
-                self._update_user_id(
-                    cluster_id,
-                    new_export.path,
-                    cast(str, new_fsal.fs_name),
-                    cast(str, new_fsal.user_id)
-                )
-                new_fsal.cephx_key = old_fsal.cephx_key
-            else:
-                expected_mds_caps = 'allow rw path={}'.format(new_export.path)
-                entity = new_fsal.user_id
-                ret, out, err = self.mgr.mon_command({
-                    'prefix': 'auth get',
-                    'entity': 'client.{}'.format(entity),
-                    'format': 'json',
-                })
-                if ret:
-                    raise NFSException(f'Failed to fetch caps for {entity}: {err}')
-                actual_mds_caps = json.loads(out)[0]['caps'].get('mds')
-                if actual_mds_caps != expected_mds_caps:
-                    self._update_user_id(
-                        cluster_id,
-                        new_export.path,
-                        cast(str, new_fsal.fs_name),
-                        cast(str, new_fsal.user_id)
-                    )
-                elif old_export.pseudo == new_export.pseudo:
-                    need_nfs_service_restart = False
-                new_fsal.cephx_key = old_fsal.cephx_key
```

Comment:
So when applying an export block if there is no change to the `path` or `fs_name` or `user_id` then `else` block here makes sure that mds caps are valid and exist as expected, I think this is an important check and must be preserved. I'm yet to reach the end of this PR but while scrambling through the patch I didn't find this being handled anywhere. So is this removal intentional? @vshankar any thoughts.

Not only this, `else` block also contain another important check:
```
elif old_export.pseudo == new_export.pseudo:
    need_nfs_service_restart = False
```
We should also ensure that this is preserved since this will make sure we don't unnecessarily restart the nfs service

Reply:
> So when applying an export block if there is no change to the `path` or `fs_name` or `user_id` then `else` block here makes sure that mds caps are valid and exist as expected, I think this is an important check and must be preserved. I'm yet to reach the end of this PR but while scrambling through the patch I didn't find this being handled anywhere. So is this removal intentional? @vshankar any thoughts.
> 
> Not only this, `else` block also contain another important check:
> 
> ```
> elif old_export.pseudo == new_export.pseudo:
>     need_nfs_service_restart = False
> ```
> 
> We should also ensure that this is preserved since this will make sure we don't unnecessarily restart the nfs service

Yes you're correct @dparmar18 .  Th echanges for `user_id` were removed because intially the assumption was to just have `cmount_path` & no requirement of `user_id`, but that's not the case now, so I'll revert these changes and preserve all this if block

Reply:
This is still kept as removed?